File: src/Java/BehavioralPatterns/Mediator/ChainReaction/src/Dropdown.java
import java.util.HashMap;
import java.util.Map;

public class Dropdown extends Participant {
    private final Map<String, Boolean> dropdownItems;

    public Dropdown(Mediator mediator) {
        super(mediator);
        dropdownItems = new HashMap<>();
        dropdownItems.put("Auto", false);
        dropdownItems.put("Manual", false);
    }

    public String getSelectedItem() {
        for (Map.Entry<String, Boolean> entry : dropdownItems.entrySet()) {
            if (entry.getValue()) {
                return entry.getKey();
            }
        }
        return null;
    }

    public void selectValue(String key) {
        String selectedDropdownItemKey = getSelectedItem();
        if (selectedDropdownItemKey != null) {
            dropdownItems.put(selectedDropdownItemKey, false);
        }
        dropdownItems.put(key, true);
        System.out.println("DropDown value changed to: " + key);
        getMediator().onStateChanged(this);
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/ChainReaction/dropdown.js

const Participant = require("./participant");

class Dropdown extends Participant {
    constructor(mediator) {
        super(mediator);

        this.dropdownItems = {
            "Auto": false,
            "Manual": false
        };
    }

    get selectedItem() {
        return Object.keys(this.dropdownItems).find(key => this.dropdownItems[key]);
    }

    selectValue(key) {
        const selectedDropdownItemKey = Object.keys(this.dropdownItems).find(key => this.dropdownItems[key]);

        if (selectedDropdownItemKey !== undefined) {
            this.dropdownItems[selectedDropdownItemKey] = false;
        }

        this.dropdownItems[key] = true;
        console.log("DropDown value changed to: " + key);
        this.mediator.onStateChanged(this);
    }
}

module.exports = Dropdown;